### ISSUE TITLE
Trigger and Semaphores

### ISSUE THREAD

**Issue by GoogleCodeExporter** [pennmush/pennmush#121]
```
I ran into an issue with @trigger... and semaphores. When using @trigger...
The object doing the @trigger becomes the enactor... See Example below:

@create Test
Created: Object #242.
@set Test=!NO_COMMAND Puppet
&COMMAND`Test_cmd test=$Test:@pemit %#=Test Passed.

@create Object
Created: Object #243.
@set Object=!NO_COMMAND Puppet
&cmd`trig Object=$Trigger:@trigger #242/COMMAND`TEST_CMD

>> Trigger
Test> You pemit "Test Passed." to Object.
Object> Test Passed.

What I propose is a way to send the player... the true enactor... the
@pemit message.

As for Semaphores. I ran into a similar problem... Where I could not pass
arguments to the string being triggered that was already in the semaphore.
The information could not be received until after the semaphore was stored.

Basically, I was making a softcoded @program command... where the part that
was being triggered was being held in a semaphore until after the player
responded with 'ANSWER foo'. The issue was... %c -> showed @pemit %#=You
typed .


My suggestion is for a way to pass on the true enactor... and 2ndly a way
to modify semaphores as needed.
```

Original issue reported on code.google.com by `adk...@drigon.com` on 13 Oct 2009 at 5:41
